treeherder/client/thclient/client.py

```diff
@@ -684,6 +686,14 @@ def _get_uri(self, project, endpoint, data=None, oauth_key=None,
 
         return uri
 
+    def _get_revision_id(self, repo_name, revision):
+        """
+        Return the id of a given revision from treeherder
+        """
+        url = "%s/%s/resultset/?revision=%s" % (self.API_ROOT, repo_name, revision)
```

Comment:
Also, you should define the resultset endpoint as a class variable, as we do with UPDATE_ENDPOINT that you can see above.